### PR TITLE
fix: More improvements via ShellCheck

### PR DESCRIPTION
**Pull request by hyperupcall**

### CODE REVIEW

File: bin/git-changelog [hunk 1]
```diff
@@ -146,6 +146,7 @@ _fetchCommitRange() {
   local start_tag="$2"
   local final_tag="$3"
 
+  # shellcheck disable=SC2086
   if [[ "$list_all" == true ]]; then
```
Comment (spacewander) [inline bin/git-changelog:150]: Why add shellcheck disable here?

Comment (hyperupcall) [reply bin/git-changelog:150]: This shellcheck comment applies to the `if` body, which contains use of variables that ate not quoted

Comment (hyperupcall) [reply bin/git-changelog:150]: There are multiple violations and the body isn't very long, which is why I put a single one at the top

Comment (spacewander) [reply bin/git-changelog:150]: > This shellcheck comment applies to the `if` body, which contains use of variables that ate not quoted

I see. Would you add a comment to indicate it's applied to the body?

Comment (hyperupcall) [reply bin/git-changelog:151]: Added the comment :+1:

**Review by spacewander — APPROVED** [review]

**Pull request merged**